libs/community/langchain_community/llms/cleanlab.py

```diff
@@ -0,0 +1,162 @@
+    and the API key set either in the ``CLEANLAB_API_KEY`` environment variable,
+    or pass it as a named parameter to the constructor.
+    Sign up at app.cleanlab.ai to get a free API key.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.llms import TrustworthyLanguageModel
+            tlm = TrustworthyLanguageModel(
+                cleanlab_api_key="my_api_key",  # Optional if `CLEANLAB_API_KEY` is set
+                quality_preset="best"
+            )
+    """
+
+    _client: Any = PrivateAttr()  # :meta private:
+
+    cleanlab_api_key: Optional[SecretStr] = Field(default=None)
+    """Cleanlab API key. Get it here: https://app.cleanlab.ai"""
+
+    quality_preset: Optional[str] = Field(default="medium")
+    """Presets to vary the quality of LLM response. Available presets listed here: 
+        https://help.cleanlab.ai/reference/python/trustworthy_language_model/#class-tlmoptions
+    """
+
+    options: Optional[Dict[str, str]] = Field(default=None)
+    """Holds configurations for trustworthy language model. 
+       Available options (model, max_tokens, etc.) with their definitions listed here: 
+       https://help.cleanlab.ai/reference/python/trustworthy_language_model/#class-tlmoptions
+    """
+
+    class Config:
+        extra = "forbid"
+
+    @model_validator(mode="before")
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        cleanlab_api_key = convert_to_secret_str(
+            get_from_dict_or_env(values, "cleanlab_api_key", "CLEANLAB_API_KEY")
+        )
+        values["cleanlab_api_key"] = cleanlab_api_key
+
+        try:
+            from cleanlab_studio import Studio
+
+            studio = Studio(api_key=cleanlab_api_key.get_secret_value())
+            # Check for user overrides in options dict
+            use_options = values["options"] is not None
+            # Initialize TLM
+            cls._client = studio.TLM(
+                quality_preset=values["quality_preset"],
+                options=values["options"] if use_options else None,
+            )
+        except ImportError:
+            raise ImportError(
+                "Could not import cleanlab-studio python package. "
+                "Please install it with `pip install -U cleanlab-studio`."
+            )
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            **{"quality_preset": self.quality_preset},
+            **{"options": self.options},
+        }
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Get the default parameters for calling Cleanlab API."""
+        default_params = {
+            "quality_preset": "medium",
+            "max_tokens": 512,
+            "model": "gpt-4o-mini",
```

Comment:
is this an openai wrapper?

Reply:
yes. TLM takes a prompt and generates a response. internally, it wraps around openai/anthropic models.

aside from generating a response (LLM), TLM also has a [method](https://help.cleanlab.ai/reference/python/trustworthy_language_model/#method-get_trustworthiness_score) which takes a prompt, and response (generated by any other LLM), to predict the `trustworthiness_score`.

i'm figuring where should this method be exposed (inside langchain), so that users can take advantage of just this functionality (simply put, `trustworthiness_score` is a metric on hallucination).

the aim is to allow users to call this method as part of a chain, and without importing `cleanlab-studio` pypi pkg (which contains TLM's `get_trustworthiness_score` method) themselves.

could you share the best practice on implementing/integrating this functionality in langchain's codebase?